catalog/watson/dialog.js

```diff
@@ -0,0 +1,63 @@
+var watson = require('watson-developer-cloud')
+var fs = require('fs')
+
+/**
+ * conversation between virtual agents and users
+ * see https://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/dialog/api/v1/
+ * param username The Watson service username.
+ * param password The Watson service password.
+ * param dialogname The name of the dialog to create.
+ * param tempfile The dialog template file used to create the dialog.
+ * param inputs The input of the user.
+ */
+function main(params) {
+    var username = params.username;
+    var password = params.password;
+    var dialogname = params.dialogname || 'my-dialog';
+    var tempfile = params.tempfile;
+    var dialogId = '';
+    var inputs = params.inputs;
+    var dialog_service = watson.dialog({
+        username: username,
+        password: password,
+        version: 'v1'
+    });
+
+    var params = {
+        name: dialogname,
+        file: fs.createReadStream(tempfile)
+    };
+
+    //create a dialog first
+    dialog_service.createDialog(params, function(err, dialog) {
```

Comment:
is the call to `createDialog` asynchronous? This is suggested by the callback. In which case `dialogId` is potentially undefined and you're starting `conversation` even though `createDialog` may not have completed or returned an error.